### DISCUSSION TITLE
Is there a reason why the Blender version is installed via the startup folder?

### DISCUSSION THREAD

**Discussion by kursad-k** [masqu3rad3/tik_manager4#84] [Q&A]
Hi,

Using the startup folder is generally a discouraged practice in Blender, all add-ons should be installed or be installable via the regular add-on installation procedures. I might maintain the Blender portion, so it would be nice to understand the initial thinking around using the startup folder. I understand why using the startup folder seem like good idea, however it is generally better idea to let Blender maintain and control the add-ons under the add-on sandboxing.

**Comment by masqu3rad3** [comment]
I cannot remember why I have chosen the startup over the addons to be fair.
I am definetely not a blender expert however I asked around people that are more familiar with Blender before starting.

I believe the current approach is one of the methods described in document, not exactly a hack. However I would be more than happy to make it better (or someone makes it better)

What would be the benefits of addons over startup scripts?

**Reply by kursad-k** [reply]
It is a good idea to let Blender maintain the add-on space for security and compatibility, also this makes it easier for the user when it comes to disabling the add-on. The startup route is generally fine for much smaller roles like setting up some paths or for basic automation, or when Blender is run in the headless mode. 

Blender Foundation soon is going to enable the extension platform which lets users download and install add-ons from an online catalog on demand. We would need the Tik Manager to be a regular add-on if we want it to be listed on that platform. This would increase the accessibility and the discoverability of the tool.

[extensions.blender.org](extensions.blender.org)

**Reply by masqu3rad3** [reply]
Very valid points.
What happens in startup script is essentially nothing more than adding paths to the python path and setting some simple commands. The core is still in an arbitrary location. Distributing with extensions platform might still be a bit tricky but would be nice indeed.
Anything related with Blender is still an uncharted territory for me but I'll try to take a look for the addon approach.

**Reply by masqu3rad3** [reply]
@kursad-k  would you like to create an issue from this discussion point? Maybe someone (like you for example :) ) with much more Blender experience can give a helping hand to this issue?

**Reply by kursad-k** [reply]
@masqu3rad3 I would be happy to take this on me. I will install and test these tools on the weekend.

**Reply by kursad-k** [reply]
Is there a TM discord/forum beside this one?

**Reply by masqu3rad3** [reply]
There is a facebook discussion group:
https://www.facebook.com/groups/tikManager
and brand new LinkedIn group:
https://www.linkedin.com/groups/13008760/

They are not much active though.
